### PR TITLE
jak2: fix warp effect for robotank and underb on other aspect ratios

### PR DESCRIPTION
**Pull request by Hat-Kid**

### CODE REVIEW

File: goal_src/jak2/levels/fortress/ammo_dump/fort-robotank-turret.gc [hunk 1]
```diff
@@ -1069,7 +1070,7 @@
        )
      (logior! (-> self flags) 128)
      (format 0 "TODO - disabling roboscreen!~%")
```
Comment (ManDude) [inline goal_src/jak2/levels/fortress/ammo_dump/fort-robotank-turret.gc:1072]: remove print.

File: goal_src/jak2/levels/fortress/ammo_dump/fort-robotank-turret.gc [hunk 2]
```diff
@@ -258,8 +258,9 @@
           (s5-0 (new 'stack-no-clear 'vector))
           (gp-0 (new 'stack-no-clear 'vector))
           (f30-0 4096.0)
+          (ratio (* (-> *pc-settings* aspect-ratio) 0.375))
           )
-      (let ((f0-4 (* 0.00013563369 (tan (* 0.5 (-> *math-camera* fov))) f30-0)))
+      (let ((f0-4 (* 0.00013563369 (tan (* ratio (-> *math-camera* fov))) f30-0)))
```
Comment (ManDude) [inline goal_src/jak2/levels/fortress/ammo_dump/fort-robotank-turret.gc:263]: tag as changed for PC.

Also, shouldn't you do `(* (-> *pc-settings* aspect-ratio-scale) 0.5)` here?

File: goal_src/jak2/levels/underport/underb-master.gc [hunk 3]
```diff
@@ -7,6 +7,9 @@
 
 ;; DECOMP BEGINS
 
+;; ratio for warp effect interp and size
+(define *underb-ratio* 0.5)
+
```
Comment (ManDude) [inline goal_src/jak2/levels/underport/underb-master.gc:12]: tag as changed for pc.

File: goal_src/jak2/levels/underport/underb-master.gc [hunk 4]
```diff
@@ -84,8 +87,13 @@
...
           (gp-0 (new 'stack-no-clear 'vector))
           (f30-0 4096.0)
+          (ratio (* (-> *pc-settings* aspect-ratio) 0.375))
           )
-      (let ((f0-4 (* 0.00013563369 (tan (* 0.5 (-> *math-camera* fov))) f30-0)))
+      (set! *underb-ratio* ratio)
+      ;; clamp interp if aspect ratio is too wide
+      (set! (-> obj interp) (if (> ratio 1.0) 0.93 ratio))
+      ;; set effect size based on aspect ratio
+      (let ((f0-4 (* 0.00013563369 (tan (* *underb-ratio* (-> *math-camera* fov))) f30-0)))
```
Comment (ManDude) [inline goal_src/jak2/levels/underport/underb-master.gc:96]: same thing as above

File: goal_src/jak2/levels/underport/underb-master.gc [hunk 5]
```diff
@@ -84,8 +87,13 @@
           (s5-0 (new 'stack-no-clear 'vector))
           (gp-0 (new 'stack-no-clear 'vector))
           (f30-0 4096.0)
+          (ratio (* (-> *pc-settings* aspect-ratio) 0.375))
           )
-      (let ((f0-4 (* 0.00013563369 (tan (* 0.5 (-> *math-camera* fov))) f30-0)))
+      (set! *underb-ratio* ratio)
+      ;; clamp interp if aspect ratio is too wide
+      (set! (-> obj interp) (if (> ratio 1.0) 0.93 ratio))
```
Comment (ManDude) [inline goal_src/jak2/levels/underport/underb-master.gc:94]: This goes from 0.99, to 1.0, then after that it clamps to 0.93? Confusing.

File: goal_src/jak2/levels/underport/underb-master.gc [hunk 6]
```diff
@@ -119,7 +127,7 @@
     (the-as skeleton-group (art-group-get-by-name *level* "skel-under-warp" (the-as (pointer uint32) #f)))
     (the-as pair 0)
     )
-  (set! (-> self interp) 0.5)
+  (set! (-> self interp) *underb-ratio*)
```
Comment (ManDude) [inline goal_src/jak2/levels/underport/underb-master.gc:130]: tag as changed for pc.

**Review by ManDude — CHANGES_REQUESTED** [review]

File: goal_src/jak2/levels/underport/underb-master.gc (hunk 5 above)
Comment (Hat-Kid) [reply goal_src/jak2/levels/underport/underb-master.gc:94]: Past 0.93, the effect gets a bit too subtle and completely stops at 1.0, so I just clamped it to this. I'll change the condition to `> ratio 0.93` for clarity.

**Review by ManDude — APPROVED** [review]

**Pull request merged**